staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt.go

```diff
@@ -115,46 +114,96 @@ type requestManagementSystem struct {
-	// pointer here, but in this case the range of the
-	// `RMState.priorityLevels` will be reduced --- by removal of the
-	// priority level that is no longer in use.
+	// curState holds a pointer to the current requestManagementState.
+	// That is, `Load()` produces a `*requestManagementState`.  When a
+	// config work queue worker processes a configuration change, it
+	// stores a new pointer here --- it does NOT side-effect the old
+	// `requestManagementState` value.  The new
+	// `requestManagementState` has a freshly constructed slice of
+	// FlowSchema pointers and a freshly constructed map of priority
+	// level states.
 	curState atomic.Value
 }
 
 // NewRequestManagementSystem creates a new instance of request-management system
 func NewRequestManagementSystem(
 	informerFactory kubeinformers.SharedInformerFactory,
+	flowcontrolClient rmclientv1alpha1.FlowcontrolV1alpha1Interface,
+	queueSetFactory fq.QueueSetFactory,
+	serverConcurrencyLimit int,
+	requestWaitLimit time.Duration,
+) Interface {
+	return NewRequestManagementSystemWithPreservation(
+		informerFactory,
+		flowcontrolClient,
+		queueSetFactory,
+		serverConcurrencyLimit,
+		requestWaitLimit,
+		nil, nil)
+}
+
+// NewRequestManagementSystemWithPreservation creates a new instance of request-management system with preservation
+func NewRequestManagementSystemWithPreservation(
+	informerFactory kubeinformers.SharedInformerFactory,
+	flowcontrolClient rmclientv1alpha1.FlowcontrolV1alpha1Interface,
+	queueSetFactory fq.QueueSetFactory,
 	serverConcurrencyLimit int,
 	requestWaitLimit time.Duration,
-	clk clock.Clock,
+	preservingFlowSchemas []*rmtypesv1alpha1.FlowSchema,
+	preservingPriorityLevels []*rmtypesv1alpha1.PriorityLevelConfiguration,
 ) Interface {
 	reqMgmt := &requestManagementSystem{
-		clk:                    clk,
-		configQueue:            workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(200*time.Millisecond, 8*time.Hour), "req_mgmt_config_queue"),
-		plLister:               informerFactory.Flowcontrol().V1alpha1().PriorityLevelConfigurations().Lister(),
-		fsLister:               informerFactory.Flowcontrol().V1alpha1().FlowSchemas().Lister(),
+		queueSetFactory:        queueSetFactory,
 		serverConcurrencyLimit: serverConcurrencyLimit,
 		requestWaitLimit:       requestWaitLimit,
+		flowcontrolClient:      flowcontrolClient,
 	}
-	// TODO: finish implementation
-	return reqMgmt
-}
+	klog.V(2).Infof("NewRequestManagementSystem with serverConcurrencyLimit=%d, requestWaitLimit=%s", serverConcurrencyLimit, requestWaitLimit)
+	reqMgmt.initializeConfigController(informerFactory)
+	emptyRMState := &requestManagementState{
+		priorityLevelStates: make(map[string]*priorityLevelState),
+	}
+	reqMgmt.curState.Store(emptyRMState)
+	if len(preservingFlowSchemas) > 0 || len(preservingPriorityLevels) > 0 {
+		reqMgmt.digestConfigObjects(
+			preservingPriorityLevels,
+			preservingFlowSchemas,
+		)
+	}
```

Comment:
`else` the filter will reject all requests until the config controller has digested enough config objects.  That's not very good behavior.